### PR TITLE
Patched the user agent core onInvite to interfere when DND is on

### PR DESCRIPTION
**Pull request by patrickswijgman**
### Issue number

#75 

### Description of fix

Patched the onInvite delegate of the UserAgentCore when instantiating a UserAgent to reject an IncomingInviteRequest early when DND is on. Of course the fix like it is only works for SIP 0.15.x because in 0.16.0 the User Agent has been refactored again but the idea should stay the same.

### CODE REVIEW

**Comment by Dagg-h** [comment]
lgtm :)

File: demo/components/c-contacts.mjs [hunk 1]
```diff
@@ -23,28 +23,32 @@ window.customElements.define(
...
-                logger.info(`Subscribing to ${accountId}`);
+            if (dataset.action) {
+              const accountId = this.nodes.input.value;
+              switch (dataset.action) {
+                case 'subscribe':
+                  {
+                    logger.info(`Subscribing to ${accountId}`);
 
-                await sipClient.subscribe(this.nodes.input.value);
+                    await sipClient.subscribe(this.nodes.input.value);
```
Comment (confususs) [inline demo/components/c-contacts.mjs:38]: You could re-use accountId here :)

File: demo/components/c-contacts.mjs [hunk 2]
```diff
@@ -23,28 +23,32 @@ window.customElements.define(
...
-                this.nodes.contactsList.appendChild(contactNode);
-                break;
-              case 'unsubscribe':
-                logger.info(`Unsubscribing from ${accountId}`);
-                sipClient.unsubscribe(accountId);
-                break;
-              default:
-                break;
+                    const contactNode = document.createElement('c-contact');
+                    contactNode.data = { contactUri: this.nodes.input.value };
```
Comment (confususs) [inline demo/components/c-contacts.mjs:41]: Same here

File: test/client-disconnect.ts [hunk 3]
```diff
@@ -170,6 +168,7 @@ test.serial('not waiting for unregistered if hasRegistered = false', async t =>
   };
 
   const client = createClientImpl(ua, defaultTransportFactory());
+  // eslint-disable-next-line require-atomic-updates
   client.disconnect = async () => {
     await (client as any).transport.disconnect({ hasRegistered: false });
```
Comment (confususs) [inline test/client-disconnect.ts:173]: If you remove the async / await and return the `transport.disconnect({..})` the linter will probably be happy & you can remove the disable-next-line

**Review by confususs — CHANGES_REQUESTED** [review]

**Comment by patrickswijgman** [comment]
Closing this in favour of https://github.com/open-voip-alliance/WebphoneLib/pull/80

**Pull request closed without merging**